common/app/conf/HealthCheck.scala

```diff
@@ -0,0 +1,127 @@
+package conf
+
+import common._
+import org.joda.time.DateTime
+import play.api.{Mode, Play}
+import play.api.libs.ws.WS
+import play.api.mvc._
+import play.api.{Application => PlayApp, GlobalSettings}
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+/*
+* Healthcheck endpoints are requested periodically
+* Results are stored in a cache
+* Goal: Requests made to the healthcheck endpoint by the ELBs are not 'inlined'
+* */
+
+private[conf] case class HealthCheckResult(url: String,
+                                           result: Either[Throwable, Int], // Status code if success, throwable otherwise
+                                           date: DateTime = DateTime.now,
+                                           expiration: Duration = 10.seconds) {
+  private val expirationDate = date.plus(expiration.toMillis)
+  private def expired: Boolean = DateTime.now.getMillis > expirationDate.getMillis
+  def recentlySucceed: Boolean = result.fold(_ => false, _ == 200) && !expired
+  def formattedResult: String = result match {
+    case Left(t) => s"Error: ${t.getLocalizedMessage}"
+    case Right(status) => status.toString
+  }
+  def formattedDate: String = if(expired) s"${date} (Expired)" else date.toString
+}
+
+private[conf] trait HealthCheckFetcher extends ExecutionContexts with Logging {
+  import play.api.Play.current
+
+  protected def fetchResult(baseUrl: String, port: Int, path: String): Future[HealthCheckResult] = {
+    WS.url(s"$baseUrl$path")
+      .withHeaders("X-Gu-Management-Healthcheck" -> "true")
```

Comment:
why not but what the difference to use the `user-agent` header rather than a custom `X-Gu-Management-Healthcheck`?